flocker/control/test/test_httpapi.py

```diff
@@ -459,23 +722,27 @@ def get_dataset_ids(deployment):
     CreateDatasetTestsMixin, "CreateDataset", _build_app)
 
 
+def _manifestation(**kwargs):
+    """
+    :param bool primary: Whether to create a primary or replica
+        ``Manifestation``. Defaults to ``True``.
+    :param kwargs: Additional keyword arguments to use to initialize the
+        manifestation's ``Dataset``.
+
+    :return: A ``Manifestation`` for a dataset with a new
+        random identifier.
+    """
+    primary = kwargs.pop('primary', True)
```

Comment:
:question: `primary` is not really a parameter. Maybe remove `:param bool primary:` and document that `kwargs` must include a boolean `primary`.